### PR TITLE
Add js/admin/customers.js

### PR DESCRIPTION
**Pull request by wvega**
# Summary

This PR adds the script to display a modal in the WooCommerce Customer's page.

## Details

To avoid browser compatibility issues, I added and configured Babel to transform our JS code into ES5.

### Setup

1. Remove Jilt for WooCommerce
1. Install Memberships
1. Update `composer.json` to use branch `dev-ch61067/customers-page-js-file` of `skyverge/wc-jilt-promotions`.
1. `composer update skyverge/wc-jilt-promotions`
1. Delete the `_sv_wc_jilt_enabled_messages` and `_sv_wc_jilt_dismissed_messages` meta entries from your user account

### Closing the modal

1. Go to the WooCommerce > Customers page
1. Click the Download button
    - [x] The installation modal is shown
    - [x] The title and content match the content for the **I want to try Jilt** modal in the [pitch](https://docs.google.com/document/d/1hm_uRWeTqrN2SiqpwGJleEBLbknZ0GsEgmVxhB8eKJQ/edit#)
    - [x] The `wc-customers-download` message was added to the `_sv_wc_jilt_enabled_messages` user meta
1. Close the modal using the X button
    - [x] The `wc-customers-download` message was added to the `_sv_wc_jilt_dismissed_messages` user meta

### Installing Jilt

1. Delete the `_sv_wc_jilt_enabled_messages` and `_sv_wc_jilt_dismissed_messages` meta entries from your user account
1. Go to the WooCommerce > Customers page
1. Click the **I want to try Jilt** button
    - [x] Jilt is installed successfully
    - [x] You're redirected to the Jilt settings page

### CODE REVIEW

File: src/Admin/Customers.php [hunk 1]
```diff
@@ -107,7 +111,7 @@ public function render_try_jilt_modal() {
 
 		?>
 		<script type="text/template" id="tmpl-sv-wc-jilt-promotions-<?php esc_attr_e( $this->download_message_id ); ?>-modal">
-			<div class="sv-wc-jilt-install-modal wc-backbone-modal">
+			<div id="sv-wc-jilt-install-modal" class="wc-backbone-modal">
 				<div class="wc-backbone-modal-content">
 					<section class="wc-backbone-modal-main" role="main">
 						<header class="wc-backbone-modal-header">
```
Comment (adrianostanley) [inline src/Admin/Customers.php:117]: In [ch61111](https://app.clubhouse.io/skyverge/story/61111/implement-render-recover-carts-modal), there's a task for creating a new view file for the modal in `views/admin`. Shouldn't this modal be placed in a view file as well?

File: src/assets/js/admin/customers.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+jQuery( document ).ready( function ($) {
+
+
+	let shouldDisplayModal = true;
+
+	// open a promotional modal when the merchant clicks the Download button in the Customers page
+	$( document ).on('click', '.woocommerce-table__download-button', function ( event ) {
```
Comment (adrianostanley) [inline src/assets/js/admin/customers.js:7]: Just a very small detail but `('click` is missing a space.

File: src/Admin/Customers.php (hunk 1 above)
Comment (wvega) [reply src/Admin/Customers.php:117]: @adrianostanley that's a good question. The initial plan was to use view files, but we switched to adding the template inline [on a previous PR](https://github.com/skyverge/wc-jilt-promotions/pull/7#pullrequestreview-465812546). We don't have a documented good practice about how to handle those, so I opted to keep it consistent when I added the modal for Customers.

Considering the above, we should probably update the task to render the modal inline.

Comment (adrianostanley) [reply src/Admin/Customers.php:117]: Great! I'll keep the one for ch61111 inline as well.

File: src/assets/js/admin/customers.js (hunk 2 above)
Comment (wvega) [reply src/assets/js/admin/customers.js]: Thanks for catching that one. Fixed in c68ad75.

**Comment by wvega** [comment]
Thank you for the reviews @adrianostanley @N-Molham. 

>I don't know if that's relevant at this point, but it looks like the link and the button are missing a margin in order to reproduce the same modal as the pitch one.

That's a good point. We will have to add a bit of CSS to make those look correct. I added CSS classes to the actions in the modal to use the CSS from [promotions.css](https://github.com/skyverge/wc-jilt-promotions/pull/20/files#diff-3d17078c3c780970aea5a7845b79e948) (see f02c3f3):

image

I'll merge this one for now, and we can make additional tweaks if necessary.

**Pull request merged**